src/storage/txn/actions/prewrite.rs

```diff
@@ -618,6 +618,12 @@ impl<'a> PrewriteMutation<'a> {
         if let Some(secondary_keys) = self.secondary_keys {
             lock.use_async_commit = true;
             lock.secondaries = secondary_keys.to_owned();
+        } else if try_one_pc && lock.primary == self.key.to_raw()? {
+            // Set use_async_commit to true when try_one_pc and the key is primary. If
+            // use_async_commit is false and async-prewrite-apply is enabled, there is a
+            // chance that reads with max_ts cannot see the previous writes because the mem
+            // lock can be skipped and the data hasn't been applied yet.
+            lock.use_async_commit = true;
```

Comment:
How about wrapping the logic into the `Lock::use_async_commit` function?

Reply:
Do you mean to wrap the condition logic? I see `Lock::use_async_commit` is used in decoding a lock from bytes, where there is no txn context (`try_one_pc` & `mutation.key`).

Reply:
Yes so the `lock.use_async_commit = true;` is wrapped into one place, but this does indeed require additional input parameters for the wrapped function.

Reply:
But `Lock::use_async_commit` is currently used to construct a lock from raw bytes (instead of a prewrite mutation), thus "additional input parameters" are unavailable when constructing from bytes.